--- a/Makefile.am
+++ b/Makefile.am
@@ -696,15 +696,15 @@
 
 install-data-hook:
 if HPLIP_BUILD
-	if [ \( "$(UNAME)" = "x86_64" -a  -d "$(libdir)/" \) ]; then \
-		cp prnt/hpcups/libImageProcessor-x86_64.so $(libdir)/ ; \
-		chmod 775 $(libdir)/libImageProcessor-x86_64.so ; \
-		ln -sf $(libdir)/libImageProcessor-x86_64.so $(libdir)/libImageProcessor.so ; \
+	if [ \( \( "$(UNAME)" = "x86_64" -o "$(UNAME)" = "aarch64" \) -a  -d "$(DESTDIR)$(libdir)/" \) ]; then \
+		cp prnt/hpcups/libImageProcessor-x86_64.so $(DESTDIR)$(libdir)/ ; \
+		chmod 775 $(DESTDIR)$(libdir)/libImageProcessor-x86_64.so ; \
+		ln -sf libImageProcessor-x86_64.so $(DESTDIR)$(libdir)/libImageProcessor.so ; \
 	fi; \
-	if [ \( \( "$(UNAME)" = "i686" -o "$(UNAME)" = "i386" \) -a -d "$(libdir)/" \) ]; then \
-		cp prnt/hpcups/libImageProcessor-x86_32.so $(libdir)/ ; \
-		chmod 775 $(libdir)/libImageProcessor-x86_32.so ; \
-		ln -sf $(libdir)/libImageProcessor-x86_32.so $(libdir)/libImageProcessor.so ; \
+	if [ \( \( "$(UNAME)" = "i686" -o "$(UNAME)" = "i386" \) -a -d "$(DESTDIR)$(libdir)/" \) ]; then \
+		cp prnt/hpcups/libImageProcessor-x86_32.so $(DESTDIR)$(libdir)/ ; \
+		chmod 775 $(DESTDIR)$(libdir)/libImageProcessor-x86_32.so ; \
+		ln -sf libImageProcessor-x86_32.so $(DESTDIR)$(libdir)/libImageProcessor.so ; \
 	fi
 	if [ -d "/usr/share/ipp-usb/quirks/" ]; then \
 		echo "ipp-usb directory exists"; \
@@ -721,15 +721,17 @@
 		  echo "Adding hpaio entry to /etc/sane.d/dll.conf." ; \
 		  echo hpaio >>$(DESTDIR)/etc/sane.d/dll.conf ; \
 	   fi; \
- 	   if [ \( "$(UNAME)" = "x86_64" -a  -d "$(libdir)/x86_64-linux-gnu/sane" \) ]; then \
-	  	ln -sf $(libdir)/sane/libsane-hpaio.so $(libdir)/x86_64-linux-gnu/sane/ ; \
-	  	ln -sf $(libdir)/sane/libsane-hpaio.so.1 $(libdir)/x86_64-linux-gnu/sane/ ; \
+ 	   if [ \( "$(UNAME)" = "x86_64" -o "$(UNAME)" = "aarch64" \) ]; then \
+		mkdir -p $(DESTDIR)$(libdir)/x86_64-linux-gnu/sane ; \
+		ln -sf $(libdir)/sane/libsane-hpaio.so $(DESTDIR)$(libdir)/x86_64-linux-gnu/sane/ ; \
+		ln -sf $(libdir)/sane/libsane-hpaio.so.1 $(DESTDIR)$(libdir)/x86_64-linux-gnu/sane/ ; \
 	   fi; \
-	   if [ \( \( "$(UNAME)" = "i686" -o "$(UNAME)" = "i386" \) -a -d "$(libdir)/i386-linux-gnu" \) ]; then \
-	      ln -sf $(libdir)/libhpmud.so.0.0.6  $(libdir)/i386-linux-gnu/libhpmud.so ; \
-	      ln -sf $(libdir)/libhpmud.so.0.0.6  $(libdir)/i386-linux-gnu/libhpmud.so.0 ; \
-	      ln -sf $(libdir)/sane/libsane-hpaio.so.1.0.0 $(libdir)/i386-linux-gnu/sane/libsane-hpaio.so.1 ; \
-	      ln -sf $(libdir)/sane/libsane-hpaio.so.1.0.0 $(libdir)/i386-linux-gnu/sane/libsane-hpaio.so ; \
+	   if [ \( "$(UNAME)" = "i686" -o "$(UNAME)" = "i386" \) ]; then \
+	      mkdir -p $(DESTDIR)$(libdir)/i386-linux-gnu/sane ; \
+	      ln -sf $(libdir)/libhpmud.so.0.0.6  $(DESTDIR)$(libdir)/i386-linux-gnu/libhpmud.so ; \
+	      ln -sf $(libdir)/libhpmud.so.0.0.6  $(DESTDIR)$(libdir)/i386-linux-gnu/libhpmud.so.0 ; \
+	      ln -sf $(libdir)/sane/libsane-hpaio.so.1.0.0 $(DESTDIR)$(libdir)/i386-linux-gnu/sane/libsane-hpaio.so.1 ; \
+	      ln -sf $(libdir)/sane/libsane-hpaio.so.1.0.0 $(DESTDIR)$(libdir)/i386-linux-gnu/sane/libsane-hpaio.so ; \
 	   fi \
 	fi
 endif #HPLIP_CLASS_DRIVER
