--- a/poppler/CertificateInfo.h
+++ b/poppler/CertificateInfo.h
@@ -70,8 +70,8 @@
     {
         PublicKeyInfo() = default;
 
-        PublicKeyInfo(PublicKeyInfo &&) noexcept = default;
-        PublicKeyInfo &operator=(PublicKeyInfo &&) noexcept = default;
+        PublicKeyInfo(PublicKeyInfo &&) = default;
+        PublicKeyInfo &operator=(PublicKeyInfo &&) = default;
 
         PublicKeyInfo(const PublicKeyInfo &) = delete;
         PublicKeyInfo &operator=(const PublicKeyInfo &) = delete;
@@ -86,8 +86,8 @@
         EntityInfo() = default;
         ~EntityInfo() = default;
 
-        EntityInfo(EntityInfo &&) noexcept = default;
-        EntityInfo &operator=(EntityInfo &&) noexcept = default;
+        EntityInfo(EntityInfo &&) = default;
+        EntityInfo &operator=(EntityInfo &&) = default;
 
         EntityInfo(const EntityInfo &) = delete;
         EntityInfo &operator=(const EntityInfo &) = delete;
